core/src/main/kotlin/org/evomaster/core/output/TestCaseWriter.kt

```diff
@@ -381,50 +387,99 @@ class TestCaseWriter {
         }
         else{
             when(resContentsItem::class) {
-                Double::class -> return "equalTo(${(resContentsItem as Double).toInt()})"
-                String::class -> return "containsString(\"${resContentsItem}\")"
+                //Double::class -> return "anyOf(equalTo(${(Math.floor(resContentsItem as Double).toInt())}), closeTo(${(resContentsItem as Double)}, 0.1))"
+                Double::class -> return "numberMatches(${resContentsItem as Double})"
+                String::class -> return "containsString(\"${(resContentsItem as String)
+                        .replace("\"", "\\\"")
+                        .replace("\n", "\\n")
+                        .replace("\r", "\\r")}\")"
+                //Note: checking a string can cause (has caused) problems due to unescaped quotation marks
+                // The above solution should be refined.
                 else -> return "NotCoveredYet"
             }
         }
+        /* BMR: the code above is due to a somewhat unfortunate problem:
+        - Gson does parses all numbers as Double
```

Comment:
a comment: JSON has no concept of integers. all numeric values are floating point precision. Swagger could specify that a number is indeed an integer, but Gson would not know about it

Reply:
I clarified the comment a bit. The swagger could specify that info, but I cannot be certain that it will. In addition, the NumberMatcher can also be used as an example of additional matchers (should they be needed in future). Clarification will become visible once I push the changes.